modular_skyrat/modules/modular_implants/code/soulcatcher/soulcatcher_component.dm

```diff
@@ -381,6 +388,67 @@ GLOBAL_LIST_EMPTY(soulcatchers)
 	room_to_join.add_soul_from_ghost(src)
 	return TRUE
 
+/**
+ * Args:
+ * datum/component/soulcatcher/soulcatcher - The soulcatcher we are checking.
+ *
+ * Returns: TRUE if we can join any room at all, FALSE otherwise.
+ */
+/mob/proc/can_join_soulcatcher(datum/component/soulcatcher/soulcatcher)
+	SHOULD_CALL_PARENT(TRUE)
+
+	var/room_joinable = FALSE
+	for(var/datum/soulcatcher_room/room as anything in soulcatcher.soulcatcher_rooms)
+		if(can_join_soulcatcher_room(room, FALSE)) // checking parent = recursion
+			room_joinable = TRUE
+			break
+
+	return room_joinable
+
+/**
+ * Args:
+ * datum/component/soulcatcher/soulcatcher - The soulcatcher we are checking.
+ *
+ * Returns: False if super returned false, false if the soulcatcher is not joinable by ghosts, true otherwise.
+ */
+/mob/dead/observer/can_join_soulcatcher(datum/component/soulcatcher/soulcatcher)
+	. = ..()
+	if (!.)
+		return FALSE
+
+	if (!isobj(soulcatcher.parent))
+		return FALSE
+
+	if (!soulcatcher.ghost_joinable)
+		return FALSE
+
+	// super returns a value by default, and we return false if it returns false, so if it reaches here - we returned true
```

Comment:
You don't have to explain anything if you just make it explicit, and you future-proof yourself in case something happens where it returns something that isn't explicitly `TRUE`.
```suggestion
	return TRUE
```